app/assets/javascripts/components/nav.jsx

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import { Link } from 'react-router';
+import { slide as Menu } from 'react-burger-menu';
+const Nav = React.createClass({
+  displayName: 'Nav',
+
+  getInitialState() {
+    const rootUrl = $('#nav_root').data('rooturl');
+    const logoPath = $('#nav_root').data('logopath');
+    const fluid = $('#nav_root').data('fluid');
+    const exploreurl = $('#nav_root').data('exploreurl');
+    const explorename = $('#nav_root').data('explorename');
+    const exploreclass = $('#nav_root').data('exploreclass');
+    const usersignedin = $('#nav_root').data('usersignedin');
+    const ifadmin = $('#nav_root').data('ifadmin');
+    const trainingurl = $('#nav_root').data('trainingurl');
+    const disableTraining = $('#nav_root').data('disable_training');
+    const helpDisabled = $('#nav_root').data('help_disabled');
+    const askUrl = $('#nav_root').data('ask_url');
+    const userPermissions = $('#nav_root').data('user_permissions');
+    const wikiEd = $('#nav_root').data('wiki_ed');
+    const languageSwitcherEnabled = $('#nav_root').data('language_switcher_enabled');
+    const currentUser = $('#nav_root').data('username');
+    const destroyUrl = $('#nav_root').data('destroyurl');
+    const omniauthUrl = $('#nav_root').data('omniauth_url');
+
+    return {
+      rootUrl: rootUrl,
+      logoPath: logoPath,
+      fluid: fluid,
+      exploreclass: exploreclass,
+      exploreurl: exploreurl,
+      explorename: explorename,
+      usersignedin: usersignedin,
+      ifadmin: ifadmin,
+      trainingurl: trainingurl,
+      disableTraining: disableTraining,
+      helpDisabled: helpDisabled,
+      askUrl: askUrl,
+      wikiEd: wikiEd,
+      userPermissions: userPermissions,
+      languageSwitcherEnabled: languageSwitcherEnabled,
+      currentUser: currentUser,
+      destroyUrl: destroyUrl,
+      omniauthUrl: omniauthUrl,
+      width: $(window).width(),
+      height: $(window).height()
+    };
+  },
+
+  componentWillMount() {
+    this.updateDimensions();
+  },
+
+  componentDidMount() {
+    window.addEventListener("resize", this.updateDimensions);
+  },
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateDimensions);
+  },
+
+  updateDimensions() {
+    this.setState({ width: $(window).width(), height: $(window).height() });
+  },
+
+  showSettings(event) {
+    event.preventDefault();
+  },
+
+  render() {
+    let navBar;
```

Comment:
This is going to be a pretty complex bunch of logic to handle all the different switches of possible links, but they are mostly independent of one another.

Whitespace, comments, and grouping of related code into isolated units will be really helpful to keep this easy to work with and refactor. (See `CourseNavbar` for a similar situation.)